File: sdk/php/src/Resources/Collocation.php
<?php

namespace HiTechCloud\SDK\Resources;

class Collocation extends BaseResource
{
    /**
     * List PDU ports
     */
    public function listPdu(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/pdu');
    }

    /**
     * Get PDU port
     */
    public function getPduPort(int $serviceId, int $port): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/pdu/' . $port);
    }

    /**
     * Set PDU port state
     */
    public function setPduPort(int $serviceId, int $port, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/pdu/' . $port, $data);
    }

}

File: sdk/csharp/Resources/Collocation.cs
using System.Collections.Generic;
using System.Threading.Tasks;

namespace HiTechCloud.SDK.Resources;

public class Collocation : BaseResource
{
    public Collocation(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> PduAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/pdu", id));
    }

    public async Task<Dictionary<string, object?>> PduPortAsync(int id, int port)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/pdu/{port}", id, port));
    }

    public async Task<Dictionary<string, object?>> TogglePduPortAsync(int id, int port, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/pdu/{port}", id, port), data);
    }
}
